addons/mail/models/mail_thread.py

```diff
@@ -1005,6 +1006,63 @@ def message_route(self, message, message_dict, model=None, thread_id=None, custo
             (email_from, email_to, message_id)
         )
 
+    def _ignore_incoming_email(self, message, message_dict, routes):
+        """This method returns True is the incoming email should be ignored.
+
+        The goal of this method is to prevent loops which can occur if an auto-replier
+        send email to Odoo.
+
+        It also detects common headers set by the auto-repliers / mailing lists.
+        """
+
+        # Detect if the email has been sent by an auto-replier or by a mailing list
+        for header in ('X-Autoreply', 'X-Autorespond', 'Feedback-ID', 'List-Id', 'List-Unsubscribe'):
+            if header in message:
+                _logger.info('Email sent by an auto-replier (%s), ignoring it.', header)
+                return True
+
+        if 'Auto-Submitted' in message and message['Auto-Submitted'].lower() != 'no':
+            _logger.info('Email sent by an auto-replier (Auto-Submitted), ignoring it.')
+            return True
+
+        if 'Precedence' in message and message['Precedence'].lower() in ('bulk', 'auto_reply', 'list'):
+            _logger.info('Email sent by an auto-replier (Precedence), ignoring it.')
+            return True
+
+        # Detect the email address sent to many emails
+        get_param = self.env['ir.config_parameter'].sudo().get_param
+        INCOMING_LIMIT_PERIOD = int(get_param('mail.incoming.limit.period', 60))
+        INCOMING_LIMIT_ALIAS = int(get_param('mail.incoming.limit.alias', 5))
+
+        create_date_limit = datetime.datetime.now() - datetime.timedelta(minutes=INCOMING_LIMIT_PERIOD)
+
+        for model, thread_id, *__ in routes or []:
+            if thread_id:
+                # Reply to an existing thread
+                continue
+
+            # Creation of a record by an alias
+            model = self.env[model]
+            primary_email = getattr(model, '_primary_email', None)
+            if not primary_email or primary_email not in model._fields:
+                _logger.info('Primary email missing on %s', model._name)
+                continue
+
+            email_from = message_dict.get('email_from')
+            if not email_from:
+                continue
+
+            mail_incoming_messages_count = model.sudo().search_count([
+                ('create_date', '>', create_date_limit),
+                (primary_email, 'ilike', tools.email_normalize(email_from)),
```

Comment:
No need to do a new model imo. We could just check if the current model inherits from `mail.thread.blacklist`, if it's the case we search on `email_normalized`, if not, we fallback to the existing code.

What do you think ? (I pushed what I have in mind)

Reply:
I think I meant "method" instead of model in my comment. Idea was to override it as it would be model-based. Indeed I was not suggesting a new model (at least I don't think :D ) .